lang/java/reef-runtime-local/src/main/java/org/apache/reef/runtime/local/driver/ContainerManager.java

```diff
@@ -151,8 +151,9 @@ Container allocateOne(final int megaBytes, final int numberOfCores) {
       final String processID = nodeId + "-" + String.valueOf(System.currentTimeMillis());
       final File processFolder = new File(this.rootFolder, processID);
       processFolder.mkdirs();
+      // setting rackName to null for now, will end up using the default one
       final ProcessContainer container = new ProcessContainer(
-          this.errorHandlerRID, nodeId, processID, processFolder, megaBytes, numberOfCores, this.fileNames, this.processObserver);
+          this.errorHandlerRID, nodeId, processID, processFolder, megaBytes, numberOfCores, null, this.fileNames, this.processObserver);
```

Comment:
We can change it for optional now, and then change it back to non-optional in the next commit. As you wish
I chose to leave it like this because it will demand fewer changes later.
The rack name will be either some that the user set or the default one. The Container will have to have a rack where it runs on.